cargo-registry-index/lib.rs

```diff
@@ -18,6 +21,24 @@ use url::Url;
 
 static DEFAULT_GIT_SSH_USERNAME: &str = "git";
 
+// We are hardcoding the github.com SSH key fingerprints from
+// https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/githubs-ssh-key-fingerprints
+// here. This is considered good enough for now, but could eventually be replaced
+// by an environment variable that acts as a `known_hosts` file.
+static GITHUB_FINGERPRINTS: Lazy<[Fingerprint; 3]> = Lazy::new(|| {
+    [
+        "SHA256:+DiY3wvvV6TuJJhbpZisF/zLDA0zPMSvHdkr4UvCOqU"
+            .parse()
+            .unwrap(),
+        "SHA256:p2QAMXNIC1TJYWeIOttrVc98/R1BUFWu3/LiyKgUfQM"
+            .parse()
+            .unwrap(),
+        "SHA256:nThbg6kXUpJWGl7E1IGOCspRomTxdCARLviKw6E5SY8"
+            .parse()
+            .unwrap(),
```

Comment:
I wouldn't pull up two dependencies just for this: the part after `SHA256:` is base64-encoded (with a missing `=` for padding), we can just decode it manually and hardcode the decoded byte array.

Reply:
good point. it looks like we already were using the `base64` crate anyway, so I went with that to make it a little more straight forward to update the fingerprints if necessary.